server/gitrest/packages/gitrest-base/src/utils/redisFs/redis.ts

```diff
@@ -66,13 +68,15 @@ export class Redis implements IRedis {
 			this.prefix = parameters.prefix;
 		}
 
-		client.on("error", (error) => {
+		redisClientConnectionManager.getRedisClient().on("error", (error) => {
```

Comment:
Is there a benefit to using `redisClientConnectionManager.getRedisClient()` every time, instead of getting it once in the constructor?

```ts
class Redis {
  private readonly client: IoRedis.default | IoRedis.Cluster;
  constructor(
    private readonly redisClientConnectionManager: IRedisClientConnectionManager,
  ) {
    this.client = redisClientConnectionManager.getRedisClient();
  }

  public async get<T>(key: string): Promise<T | undefined> {
    const stringValue = await this.client.get(this.getKey(key));
    // ...
  }
```

Reply:
If there is a strict need to call `getRedisClient` every time, like if an old client can be invalidated and then a new client created silently, which I think might be the case for MSI, then we should probably enforce that behavior with unit tests.

Reply:
```ts
export class TestRedisClientConnectionManagerWithInvalidation implements IRedisClientConnectionManager {
    private readonly mockRedisClient: Redis.Redis;
	private readonly options: Redis.RedisOptions;

	constructor(options?) {
		this.options = options;
	}

	public getRedisClient(): Redis.Redis {
		return this.mockRedisClient;
	}

    private createNewRedisClient(): void {
	    this.mockRedisClient: Redis.Redis = this.options
			? new RedisMock(this.options)
			: new RedisMock();
    }

    public invalidateClient(): void {
      this.mockRedisClient.disconnect(); // or dispose or whatever
      this.createNewRedisClient();
    }
}

// .spec.ts
const redis = new TestRedisClientConnectionManagerWithInvalidation(...);
const redisFs = new RedisFs(..., redis, ...);
it("uses fresh redis client every time", () => {
  // assert initial connection works
  redisFs.get(...);
  redis.invalidateClient();
  // assert that connection continues to work after initial connection is invalidated
  redisFs.get(...);
});
```

Reply:
Actually, per this comment, if the internal client instance changes, then the `getRedisClient().on("error", ...)` listener will stop working. It's not important, because it's just for telemetry, but worth thinking about.

Reply:
I can add this unit-test to routerlicious to the test-utils package itself. I can add the unit tests to this PR itself along with the update to the `TestRedisClientConnectionManager`. The change should not impact the unit tests in Gitrest imo. Lemme know what you think of this.

Reply:
These tests would be validating _use_ of RedisConnectionManager, not its implementation, so there would have to be tests everywhere that it's used, not just R11s. Totally fine as follow-up